dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,633 @@
+            // In MySQL and SQL Server re-creating the index is NOT necessary as the regular index
+            // is created separately
+            if (DbConnectionFactory.isPostgres() || DbConnectionFactory.isOracle()) {
+                dc.executeStatement(createTableIndex());
+            }
+        } catch (SQLException e) {
+            throw new DotRuntimeException(
+                    "An error occurred when updating the index in 'workflow_scheme_x_structure' table.",
+                    e);
+        }
+    }
+
+    private void updateWorkflowActionData(final DotConnect dc) throws DotDataException {
+
+        Logger.info(this, "Associating Workflow Actions to Workflow Schemes.");
+
+        final List<Map<String, Object>> schemeIds =
+                dc.setSQL(selectSchemeIdsForActions()).loadObjectResults();
+        schemeIds.stream().forEach(row -> {
+
+            dc.setSQL(updateSchemeIdsForActions());
+            dc.addParam(row.get("scheme_id").toString());
+            //dc.addParam(this.isLocked(row.get("requires_checkout"))? WorkflowAction.LOCKED:WorkflowAction.UNLOCKED);
+            dc.addParam(row.get("action_id").toString());
+
+            try {
+                dc.loadResult();
+            } catch (DotDataException e) {
+                throw new DotRuntimeException(
+                        "An error occurred when associating Workflow Actions to Workflow Schemes.",
+                        e);
+            }
+        });
+    } // updateWorkflowActionData.
+
+    private void addWorkflowActionStepData(final DotConnect dc) throws DotDataException {
+
+        Logger.info(this, "Adding data to 'workflow_action_step' table.");
+        final List<Map<String, Object>> results =
+                dc.setSQL(selectActionsAndSteps()).loadObjectResults();
+
+        results.stream().forEach(row -> {
+
+            dc.setSQL(insertActionsAndSteps());
+            dc.addParam(row.get("action_id"));
+            dc.addParam(row.get("step_id"));
+            dc.addParam(row.get("my_order"));
+
+            Logger.debug(this, "Adding to workflow_action_step the row: " + row);
+
+            try {
+                dc.loadResult();
+            } catch (DotDataException e) {
+
+                Logger.error(this, "ERROR on Adding to workflow_action_step the row: "
+                        + row + ", err:" + e.getMessage(), e);
+                throw new DotRuntimeException(
+                        "An error occurred when adding data to the 'workflow_action_step' table.",
+                        e);
+            }
+        });
+    } // addWorkflowActionStepData.
+
+    private void addRequiresCheckoutOptionColumn(final DotConnect dc) throws DotDataException {
+
+        boolean needToCreate = false;
+        Logger.info(this, "Adding new 'requires_checkout_option' column to 'workflow_action' table.");
+
+        try {
+            dc.setSQL(findRequiresCheckoutOptionsColumn()).loadObjectResults();
+        } catch (Throwable e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)